sweep/store.go

```diff
@@ -31,16 +34,91 @@ var (
 	byteOrder = binary.BigEndian
 
 	errNoTxHashesBucket = errors.New("tx hashes bucket does not exist")
+
+	// ErrTxNotFound is returned when querying using a txid that's not
+	// found in our db.
+	ErrTxNotFound = errors.New("tx not found")
 )
 
+// TxRecord specifies a record of a tx that's stored in the database.
+type TxRecord struct {
+	// Txid is the sweeping tx's txid.
+	Txid chainhash.Hash
```

Comment:
Can use `tlv.RecordT` here now to declare the types inline. These are also all primitive records, so no extra code would be needed.

Reply:
This is not part of the serialized tlv stream but a comment which highlights this would be good

Reply:
I wonder if the design could be improved, because we don't really need the Txid only as a key saving the record, so maybe supply it as a function parameter rather than having it as part of the record ?